test/cri-containerd/jobcontainer_test.go

```diff
@@ -326,3 +327,79 @@ func Test_RunContainer_HostVolumes_JobContainer_WCOW(t *testing.T) {
 		t.Fatalf("expected volumes to be the same within job process container. got %q but expected %q", hostStdout, containerStdout)
 	}
 }
+
+func Test_RunContainer_LocalService(t *testing.T) {
+	requireFeatures(t, featureWCOWProcess, featureHostProcess)
+
+	pullRequiredImages(t, []string{imageWindowsNanoserver})
+	client := newTestRuntimeClient(t)
+
+	podctx := context.Background()
+	sandboxRequest := getJobContainerPodRequestWCOW(t)
+
+	podID := runPodSandbox(t, client, podctx, sandboxRequest)
+	defer removePodSandbox(t, client, podctx, podID)
+	defer stopPodSandbox(t, client, podctx, podID)
+
+	containerRequest := getJobContainerRequestWCOW(t, podID, sandboxRequest.Config, imageWindowsNanoserver, nil)
+	ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
+	defer cancel()
+
+	containerRequest.Config.Annotations["microsoft.com/hostprocess-inherit-user"] = "false"
+	containerRequest.Config.Windows.SecurityContext.RunAsUsername = "nt authority\\Local Service"
+	containerID := createContainer(t, client, ctx, containerRequest)
+	defer removeContainer(t, client, ctx, containerID)
+	startContainer(t, client, ctx, containerID)
+	defer stopContainer(t, client, ctx, containerID)
+}
+
+func Test_RunContainer_VolumeMount(t *testing.T) {
+	requireFeatures(t, featureWCOWProcess, featureHostProcess)
+
+	pullRequiredImages(t, []string{imageWindowsNanoserver})
+	client := newTestRuntimeClient(t)
+
+	podctx := context.Background()
+	sandboxRequest := getJobContainerPodRequestWCOW(t)
+
+	podID := runPodSandbox(t, client, podctx, sandboxRequest)
+	defer removePodSandbox(t, client, podctx, podID)
+	defer stopPodSandbox(t, client, podctx, podID)
+
+	dir, err := ioutil.TempDir("", "example")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(dir)
+
+	tmpfn := filepath.Join(dir, "tmpfile")
+	_, err = os.Create(tmpfn)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	mounts := []*runtime.Mount{
+		{
+			HostPath:      dir,
+			ContainerPath: "/path/in/container",
```

Comment:
you could add tests for drive letters (supported) and named pipes (not supported)

Reply:
good idea

Reply:
Well truth be told, the test here currently uses a drive letter, as dir will have the full path where the temp dir was created. But agreed on the namedpipe test, and I guess we can add a test for a path with no drive letter

Reply:
Done. The container start will fail if the mount validation doesn't go through so that was kind of a weird test to make, I added a local unit test to the package instead. I did add a drive letter, no drive letter, and a single file mapping test however.